html_tag_collector/RootURLCache.py

```diff
@@ -28,17 +29,42 @@ def save_cache(self):
             json.dump(self.cache, f, indent=4)
 
     def get_title(self, url):
+        if not url.startswith('http'):
+            url = "https://" + url
+
         parsed_url = urlparse(url)
         root_url = f"{parsed_url.scheme}://{parsed_url.netloc}"
+        headers = {
+            # Some websites refuse the connection of automated requests, setting the User-Agent will circumvent that
+            "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36",
+        }
 
         if root_url not in self.cache:
             try:
-                response = requests.get(root_url)
-                soup = BeautifulSoup(response.text, 'html.parser')
-                title = soup.find('title').text
-                self.cache[root_url] = title
-                self.save_cache()
-                return title
+                response = requests.get(root_url, headers=headers, timeout=120)
+            except (requests.exceptions.SSLError, ssl.SSLError):
+                # This error is raised when the website uses a legacy SSL version, which is not supported by requests
+                # Retry without SSL verification
+                try:
```

Comment:
Some of this code logic is similarly used in collector.py (in the `get_response()` function). I think it may be a good idea to extract that logic to a function -- if not in this PR, then in a subsequent PR identified by an issue. If we've already created this code once, I see no need to repeat it.

Reply:
In this case, I think extracting the code for the user agent string, as well as the code for getting a response that accounts for a possible legacy SSL version.

Reply:
This one will be more difficult to implement since the code in `get_response()` is utilizing `async/await` and a `requests-html` session, rather than a non-async `requests` in `get_title()`. Not to mention `get_response()` has slightly different error-handling logic. Not sure it's worth implementing -- at least not at this very moment, it would probably take an overhaul of the `get_title()` function to make it `async`, but not with any of the benefits of asynchronous processing since the concurrent updates to the RootURLCache causes an error at times